src/main/java/uk/gov/hmcts/reform/prl/services/pin/C100CaseInviteService.java

```diff
@@ -27,31 +29,43 @@ public class C100CaseInviteService implements CaseInviteService {
     @Autowired
     private LaunchDarklyClient launchDarklyClient;
 
-    private CaseInvite generateCaseInvite(Element<PartyDetails> partyDetails) {
-        return new CaseInvite().generateAccessCode(partyDetails.getValue().getEmail(), partyDetails.getId(), YesOrNo.No);
+    private CaseInvite generateCaseInvite(Element<PartyDetails> partyDetails, YesOrNo isApplicant) {
+        return new CaseInvite().generateAccessCode(partyDetails.getValue().getEmail(), partyDetails.getId(), isApplicant);
     }
 
     private void sendCaseInvite(CaseInvite caseInvite, PartyDetails partyDetails, CaseData caseData) {
         caseInviteEmailService.sendCaseInviteEmail(caseInvite, partyDetails, caseData);
     }
 
     @Override
-    public CaseData generateAndSendRespondentCaseInvite(CaseData caseData) {
+    public CaseData generateAndSendCaseInvite(CaseData caseData) {
         List<Element<CaseInvite>> caseInvites = caseData.getCaseInvites() != null ? caseData.getCaseInvites() : new ArrayList<>();
 
-        log.info("Generating case invites and sending notification to respondents with email address present");
+        log.info("Generating case invites and sending notification to applicants/respondents with email address present");
 
         for (Element<PartyDetails> respondent : caseData.getRespondents()) {
-            if (!respondentHasLegalRepresentation(respondent.getValue()) && Yes.equals(respondent.getValue().getCanYouProvideEmailAddress())) {
-                CaseInvite caseInvite = generateCaseInvite(respondent);
+            if (!hasLegalRepresentation(respondent.getValue()) && Yes.equals(respondent.getValue().getCanYouProvideEmailAddress())) {
+                CaseInvite caseInvite = generateCaseInvite(respondent, No);
                 caseInvites.add(element(caseInvite));
                 sendCaseInvite(caseInvite, respondent.getValue(), caseData);
             }
         }
+        //PRLC100-431 - Generate case invites & send notification to c100 applicants for case created/submitted by citizen
+        if (launchDarklyClient.isFeatureEnabled("generate-ca-citizen-applicant-pin")
+            && CaseCreatedBy.CITIZEN.equals(caseData.getCaseCreatedBy())) {
+            log.info("Generating case invites and sending notification to citizen applicants with email");
+            for (Element<PartyDetails> applicant : caseData.getApplicants()) {
+                if (!hasLegalRepresentation(applicant.getValue()) && Yes.equals(applicant.getValue().getCanYouProvideEmailAddress())) {
+                    CaseInvite caseInvite = generateCaseInvite(applicant, Yes);
+                    caseInvites.add(element(caseInvite));
+                    sendCaseInvite(caseInvite, applicant.getValue(), caseData);
+                }
+            }
+        }
```

Comment:
when are we inviting respondents? when a case is created? in solicitor journey it happens during prepare for hearing state. if this is the requirement, fine. please check otherwise.

Reply:
@AlokDatta This is same place where respondents are invited during prepare for hearing.
For applicants as well we need to do the same so added code here to generate access PIN generation with LD flag & conditions to filter only cases created by citizen.